airflow-core/src/airflow/models/baseoperator.py

```diff
@@ -29,9 +29,10 @@
 from collections.abc import Collection, Iterable, Iterator
 from datetime import datetime
 from functools import singledispatchmethod
-from typing import TYPE_CHECKING, Any
+from typing import TYPE_CHECKING, Any, cast
 
 from sqlalchemy import select
+from sqlalchemy.orm import Session
```

Comment:
Nit: why was this moved outside of a type checking block?

Reply:
Oh, singledispatch needs it to evaluate the types, even though it only looks at the type of the first one. Gotcha.